pandas/util/testing.py

```diff
@@ -1083,6 +1084,32 @@ def _raise(left, right, err_msg):
     return True
 
 
+def assert_extension_array_equal(left, right):
+    """Check that left and right ExtensionArrays are equal.
+
+    Parameters
+    ----------
+    left, right : ExtensionArray
+        The two arrays to compare
+
+    Notes
+    -----
+    Missing values are checked separately from valid values.
+    A mask of missing values is computed for each and checked to match.
+    The remaining all-valid values are cast to object dtype and checked.
+    """
```

Comment:
Why was this needed only now? (wasn't the missing values the reason you added `assert_frame_equal` et al as overridable class methods?)

Reply:
`factorize` is I think the first case where we have a public method returning an ExtensionArray (`unique`).

I'll see if any old tests can make use of this.

Reply:
I don't see any other cases where we could apply `assert_extension_array_equal`.

The closes would be the `__getitem__` tests, but in that case we don't have an expected array to compare to, since we're testing getitem / take / etc.